app/assets/javascripts/components/timeline/week.jsx

```diff
@@ -76,7 +77,12 @@ const Week = createReactClass({
     const dateCalc = new DateCalculator(this.props.timeline_start, this.props.timeline_end, this.props.index, { zeroIndexed: false });
     let weekDatesContent;
     let meetDates;
-    if (this.props.meetings && this.props.meetings.length > 0) {
+    let meetingDate;
+    if(this.props.meetings){
+      const dateExtractor = new ExtractDate(this.props.meetings[0]);
+      meetingDate = dateExtractor.getDate();
+    }
+    if (this.props.meetings && this.props.meetings.length > 0 && meetingDate < dateCalc.end()) {
```

Comment:
I wonder if it makes more sense to do this somewhere else, such as the weekMeetings function in course_date_utils.js?

I don't like the idea of extracting the date from a string, when we've already generated this data structure by starting from dates and converting them into strings.

Reply:
Actually, the issue is that the function responsible for converting the date to a string doesn't currently take the weeks (which contain the start and end dates for each week) as props. Because of this, I need to perform this check outside the date-to-string conversion function. I'll try to implement this check in the weekMeetings function within course_date_utils.js instead of writing it seperately .

Reply:
Are you still working on this change?

Reply:
Yuppp..